goalc/main.cpp

```diff
@@ -79,7 +79,13 @@ int main(int argc, char** argv) {
     return 1;
   }
 
-  setup_logging();
+  try {
+    setup_logging();
```

Comment:
minor, but in the future I'd prefer the pattern of returning a boolean indicating success or failure, rather than throwing and requiring the user to try/catch. I know it's annoying here because a lot of the file-related functions throw (which was my original mistake).